drivers/can/can_handlers.c

```diff
@@ -44,6 +47,25 @@ Z_SYSCALL_HANDLER(can_attach_msgq, dev, msgq, filter) {
 				     (const struct zcan_filter *) filter);
 }
 
+Z_SYSCALL_HANDLER(can_attach_workq, dev, work_q, work, callback, callback_arg,
+		  filter) {
+
+	Z_OOPS(Z_SYSCALL_OBJ(dev, K_OBJ_DRIVER_CAN));
+
+	Z_OOPS(Z_SYSCALL_MEMORY_READ((struct zcan_filter *)filter,
+				     sizeof(struct zcan_filter)));
+	Z_OOPS(Z_SYSCALL_MEMORY_READ((void *)callback_arg, sizeof(void *)));
+
+	Z_OOPS(Z_SYSCALL_OBJ(work_q, K_OBJ_MSGQ));
```

Comment:
This is obviously wrong. But which K_OBJ_ is the right one here? I can't find K_OBJ_WORKQ @andrewboie

Reply:
a k_work_q is just a k_queue and a k_thread. we don't do anything to the thread, we are just going to be adding items to the k_queue object. so it's enough to do Z_OOPS(Z_SYSCALL_OBJ(&work_q->queue, K_OBJ_QUEUE).